lib/inferno/repositories/results.rb

```diff
@@ -140,6 +140,18 @@ def pass_waiting_result(result_id, message = nil)
         update(result_id, result: 'pass', result_message: message)
       end
 
+      def fail_waiting_result(result_id, message = nil)
```

Comment:
https://github.com/inferno-framework/inferno-core/pull/446/commits/80b77b82d2a9d53470a0159d42e21c36aa6801e0 While at it, I renamed the method to `update_result` since its behavior is not specific to a waiting result